client/client.cpp

```diff
@@ -63,21 +63,27 @@ void init() {
   client::BW::data::init();
 }
 
+//============================= LIFECYCLE ====================================
+
 Client::Client() : state_(new State()) {}
 
 Client::~Client() {
   state_->decref();
 }
 
-bool Client::connect(const std::string& hostname, int port) {
+//============================= OPERATIONS ===================================
+
+bool Client::connect(const std::string& hostname, int port,
+    int send_timeout_ms /* = -1 */, int receive_timeout_ms /* = -1 */) {
```

Comment:
+1 on underscore. CamelCase is mostly a convention used on Windows, but as you can see I tried to stick to underscore in BWEnv too :)